apps/challenges/views.py

```diff
@@ -2239,3 +2245,292 @@ def get_challenge_evaluation_cluster_details(request, challenge_pk):
+        error_messages.append(message)
+
+    if len(error_messages):
+        response_data["error"] = "\n".join(error_messages)
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    serializer = ZipChallengeSerializer(
+        data=yaml_file_data,
+        context={
+            "request": request,
+            "challenge_host_team": challenge_host_team,
+            "image": challenge_image_file,
+            "evaluation_script": challenge_evaluation_script_file,
+        },
+    )
+    if not serializer.is_valid():
+        message = "ERROR: Challenge metadata has invalid values"
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # Check for Challenge phases
+    challenge_phases_data = yaml_file_data.get("challenge_phases")
+    if not challenge_phases_data:
+        message = "ERROR: No challenge phase key found. Please add challenge phases in YAML file and try again!"
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    for data in challenge_phases_data:
+        test_annotation_file = data.get("test_annotation_file")
+        if test_annotation_file:
+            test_annotation_file_path = join(
+                challenge_config_location,
+                test_annotation_file,
+            )
+
+            if isfile(test_annotation_file_path):
+                challenge_test_annotation_file = read_file_data_as_content_file(test_annotation_file_path,
+                                                                                "rb", test_annotation_file_path)
+            else:
+                message = (
+                    "ERROR: No test annotation file found in zip file"
+                    "for challenge phase {}".format(data["name"])
+                )
+                error_messages.append(message)
+        else:
+            message = (
+                "ERROR: There is no key for test annotation file for"
+                "challenge phase {}.".format(data["name"])
+            )
+            error_messages.append(message)
+
+    phase_ids = []
+    for data in challenge_phases_data:
+        is_valid, message = is_challenge_phase_config_yaml_html_field_valid(data,
+                                                                            "description",
+                                                                            challenge_config_location)
+        if not is_valid:
+            error_messages.append(message)
+
+        serializer = ChallengePhaseCreateSerializer(
+            data=data,
+            context={
+                "exclude_fields": ["challenge"],
+                "test_annotation": challenge_test_annotation_file,
+            },
+        )
+        if not serializer.is_valid():
+            message = "ERROR: Challenge phase {} has schema errors".format(data["id"])
```

Comment:
Let's be precise with the error message here.

Reply:
should we show the serializer errors directly? because when i tried to read those messages from a challenge host point of view I couldn't understand those. They were standard django developer errors